File: stress/t/ctl_collatz.c
#include <stdio.h>
#include <stdint.h>
int main(void){
  int64_t start=0,n=0,steps=0,totsteps=0,maxsteps=0,maxstart=0;
  for(start=1;start<=27;start=start+1){
    n=start;
    steps=0;
    while(n!=1 && steps<1000){
      if(n%2==0){ n=n/2; } else { n=3*n+1; }
      steps=steps+1;
    }
    totsteps=totsteps+steps;
    if(steps>maxsteps){ maxsteps=steps; maxstart=start; } else { maxsteps=maxsteps; }
  }
  printf("totsteps="); printf("%ld",(long)totsteps); printf("\n");
  printf("maxsteps="); printf("%ld",(long)maxsteps); printf("\n");
  printf("maxstart="); printf("%ld",(long)maxstart); printf("\n");
  return 0;
}

File: stress/t/ctl_collatz.f
      PROGRAM CTLCOLL
      INTEGER*8 START,N,STEPS,TOT,MAXS,MAXST
      TOT=0
      MAXS=0
      MAXST=0
      START=1
   10 IF (START.GT.27) GOTO 40
      N=START
      STEPS=0
   20 IF ((N.EQ.1).OR.(STEPS.GE.1000)) GOTO 30
      IF (MOD(N,2_8).EQ.0) THEN
        N=N/2
      ELSE
        N=3*N+1
      END IF
      STEPS=STEPS+1
      GOTO 20
   30 CONTINUE
      TOT=TOT+STEPS
      IF (STEPS.GT.MAXS) THEN
        MAXS=STEPS
        MAXST=START
      END IF
      START=START+1
      GOTO 10
   40 CONTINUE
      WRITE(*,'(A,I0)') 'totsteps=',TOT
      WRITE(*,'(A,I0)') 'maxsteps=',MAXS
      WRITE(*,'(A,I0)') 'maxstart=',MAXST
      END
